### PR TITLE
Moved transforms off reactive components

### PR DESCRIPTION
**Pull request by samiwh**
#### Description
Removed reactive components from transforms.

### CODE REVIEW

File: workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DirectTransformUpdateSystem.cs [hunk 1]
```diff
@@ -19,26 +15,48 @@ namespace Improbable.Gdk.TransformSynchronization
...
+            var transformToSetArray = transformGroup.GetComponentDataArray<TransformToSet>();
+            var spatialEntityIdArray = transformGroup.GetComponentDataArray<SpatialEntityId>();
+            var transformComponentArray = transformGroup.GetComponentDataArray<TransformInternal.Component>();
+
+            for (int i = 0; i < transformToSetArray.Length; ++i)
             {
-                var t = data.Transform[i];
-                data.TransfomToSet[i] = new TransformToSet
+                var entityId = spatialEntityIdArray[i];
+                // todo this means an api is missing
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DirectTransformUpdateSystem.cs]: What API is missing?

File: workers/unity/Packages/com.improbable.gdk.transformsynchronization/MonoBehaviours/TransformSynchronization.cs [hunk 2]
```diff
@@ -94,6 +94,7 @@ private void ApplyStrategies()
             if (SetKinematicWhenNotAuthoritative)
             {
                 commandBuffer.AddComponent(entity, new ManageKinematicOnAuthorityChangeTag());
+                commandBuffer.AddComponent(entity, new KinematicStateWhenAuth());
```
Comment (jamiebrynes7) [inline workers/unity/Packages/com.improbable.gdk.transformsynchronization/MonoBehaviours/TransformSynchronization.cs]: Is this a driveby bugfix?

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.transformsynchronization/MonoBehaviours/TransformSynchronization.cs]: Yes. Noticed when testing that nothing happened with the kinematic state. Eventually found out I'd not wired it up.

File: workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DirectTransformUpdateSystem.cs (hunk 1 above)
Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DirectTransformUpdateSystem.cs]: Which entities have updates on them. Ultimately I think transforms might just need their own semi-reactive component.

Comment (samiwh) [reply workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/DirectTransformUpdateSystem.cs]: Will remove the comment though.
